wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustomevent/Cron.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+/**
+ * @file
+ */
+
+use Civi\Api4\EckEntity;
+use Civi\Api4\MessageTemplate;
+use Civi\Api4\OptionValue;
+
+/**
+ * Civirules.Cron API specification.
+ *
+ * @param array $spec
+ *   Description of fields supported by this API call.
+ *
+ * @return void
+ */
+function _civicrm_api3_goonjcustomevent_cron_spec(&$spec) {
+  // No parameters for the civirules cron.
+}
+
+/**
+ * Civirules.Cron API.
+ *
+ * @param array $params
+ *
+ * @return array API result descriptor
+ *
+ * @throws \CRM_Core_Exception
+ */
+function civicrm_api3_goonjcustomevent_cron($params) {
+  $returnValues = [];
+
+  try {
+    goonjcustomevent_check_and_send_emails_for_camp_end_date();
+  }
+  catch (Exception $e) {
+    error_log('Goonj Cron Job: Error - ' . $e->getMessage());
+  }
+
+  return civicrm_api3_create_success($returnValues, $params, 'Goonjcustomevent', 'cron');
+}
+
+/**
+ * Function to check camp end date and send emails.
+ */
+function goonjcustomevent_check_and_send_emails_for_camp_end_date() {
+  $optionValues = OptionValue::get(FALSE)
+    ->addWhere('option_group_id:label', '=', 'ECK Subtypes')
+    ->addWhere('label', '=', 'Collection Camp')
+    ->execute();
+
+  $collectionCampSubtype = $optionValues->first()['value'];
+  try {
+    $collectionCamps = EckEntity::get('Collection_Camp', TRUE)
+      ->addSelect('Logistics_Coordination.Camp_to_be_attended_by', 'Collection_Camp_Intent_Details.End_Date')
+      ->addWhere('subtype', '=', $collectionCampSubtype)
+      ->execute();
+
+    // Get today's date in the same format as the End_Date field.
+    $today = new DateTime();
+    $todayFormatted = $today->format('Y-m-d');
+
+    foreach ($collectionCamps as $camp) {
+      if (empty($camp['Logistics_Coordination.Camp_to_be_attended_by'])) {
+        continue;
+      }
+      $recipientId = $camp['Logistics_Coordination.Camp_to_be_attended_by'];
+      $endDate = new DateTime($camp['Collection_Camp_Intent_Details.End_Date']);
+      $endDateFormatted = $endDate->format('Y-m-d');
+
+      // Only send the email if the end date is exactly today.
+      if ($endDateFormatted === $todayFormatted) {
+        emailToCampAttendBy($recipientId, $endDateFormatted);
+      }
+    }
+  }
+  catch (Exception $e) {
+    error_log('Error in cron job: ' . $e->getMessage());
+  }
+}
+
+/**
+ * Function to send email to camp attendee.
+ */
+function emailToCampAttendBy($contactId, $collectionCampEndDate) {
+  $messageTemplates = MessageTemplate::get(TRUE)
+    ->addSelect('id')
+    ->addWhere('msg_title', '=', 'Event Completion Notification')
```

Comment:
I have a question—why did we use a variable here instead of directly using the title? In the previous file, we did this for subtypes, which is why we added the mapper. However, I don’t think that's necessary in this case since we can achieve the same result by simply passing the name. 

Later, this will change, and we won’t need to use the template, as we created a custom template from code to pass the collection camp ID and user ID in the URL.

```->addWhere('msg_title', 'LIKE', $msgTitleStartsWith)```

Reply:
THe point was you don't need to use a variable but rather to use `LIKE` instead of the `=`. This is done because there is no way to map message templates (add custom fields) to know when to trigger which template. So what we have done is a convention for naming the message template. The starting name of the message template is hardcode value and the remaining part can be changed by the admins. So in this case it can be:

`Collection_Camp_Complete email`  where `Collection_Camp_Complete email` this part can be hardcoded